activesupport/lib/active_support/number_helper/number_to_rounded_converter.rb

```diff
@@ -42,14 +42,21 @@ def convert
       end
 
       private
+        def truncate_whole_number
+          options[:truncate_whole_number]
+        end
+
         def strip_insignificant_zeros
           options[:strip_insignificant_zeros]
         end
 
         def format_number(number)
+          escaped_separator = Regexp.escape(options[:separator])
+
           if strip_insignificant_zeros
-            escaped_separator = Regexp.escape(options[:separator])
             number.sub(/(#{escaped_separator})(\d*[1-9])?0+\z/, '\1\2').sub(/#{escaped_separator}\z/, "")
+          elsif truncate_whole_number && number.match?(/(#{escaped_separator})(0+\z)/)
```

Comment:
What happens if both of these options are set?

Reply:
@zzak Thanks for reviewing! That's a good point, right now I don't think it covers that case well (and only falls into the first `if`) so I'll adjust the code. I'll write a test and see how it plays out as well 🤔 


The other thing that I am curious about is how ergonomic it feels. It would feel weird to me if someone said `number_to_currency(strip_insiginificant_zeros: true, truncate_whole_number: true)`. So I think it's the case that only one of them should ever be at play. Additionally I think changing it to be `convert_to_whole_number` makes more sense.

Reply:
@zzak tests added! The two can work in conjunction. If I pass in 120.00 with `strip_insignificant_zeros: true` and `truncate_trailing_zeros: true`, then I still get `$120`, since that's already the behavior of `strip_insignificant_zeros` today. However `strip_insignificant_zeros` takes precedence, so if I pass in `120.05000` and `strip_insignificant_zeros: true`, `truncate_trailing_zeros: true` then I still keep `$120.05000`. I tried to solve this with the parameter name `_trailing_zeros` 🤔 

I also thought about the name more, since truncation is a bit [different than rounding](https://www.ualberta.ca/computing-science/media-library/teaching-resources/java/truncation-rounding.html).

I see a few options here actually:
1. Push for `truncate_trailing_zeros` which I've kept for now.
2. Instead rename it to `truncate` which obviates `strip_insignificant_zeros` if it's ever passed in and does no rounding (but I think that makes things a bit trickier with regards to how `precision` is handled too). In fact, we would end up ignoring `precision` if it's passed in.

The tricky thing from a user perspective of the API is that it's an options hash, so it's hard to communicate to the user if anything overrules another.

Option 2 seems a bit more complicated, and I am not sure if we want just yet.